packages/eslint-plugin/src/util/misc.ts

```diff
@@ -25,6 +25,11 @@ export function isDefinitionFile(fileName: string): boolean {
       return true;
     }
   }
+
+  if (lowerFileName.includes('.d.') && lowerFileName.endsWith('.ts')) {
+    return true;
+  }
+
   return false;
```

Comment:
[Bug] This isn't quite the same logic as the original TypeScript source: https://github.com/microsoft/DefinitelyTyped-tools/blob/1f6e243b03e6d557bf43e6312b0b8f2d47819e52/packages/utils/src/miscellany.ts#L97. Let's go with the same thing they use to play it safe:

```suggestion
  return /\.d\.(ts|cts|mts|.*\.ts)$/.test(lowerFileName);
```